chia/wallet/wallet.py

```diff
@@ -241,59 +242,34 @@ def add_condition_to_solution(self, condition: Program, solution: Program) -> Pr
 
-    async def select_coins(self, amount, exclude: List[Coin] = None) -> Set[Coin]:
+    async def select_coins(self, target: uint64, exclude: List[Coin] = None) -> Set[Coin]:
         """
         Returns a set of coins that can be used for generating a new transaction.
-        Note: This must be called under a wallet state manager lock
+        Note: Must be called under wallet state manager lock
         """
-        if exclude is None:
-            exclude = []
-
-        spendable_amount = await self.get_spendable_balance()
-
-        if amount > spendable_amount:
-            error_msg = (
-                f"Can't select amount higher than our spendable balance.  Amount: {amount}, spendable: "
-                f" {spendable_amount}"
-            )
-            self.log.warning(error_msg)
-            raise ValueError(error_msg)
-
-        self.log.info(f"About to select coins for amount {amount}")
-        unspent: List[WalletCoinRecord] = list(
+        spendable_amount: uint128 = await self.get_spendable_balance()
+        spendable_coins: List[WalletCoinRecord] = list(
             await self.wallet_state_manager.get_spendable_coins_for_wallet(self.id())
         )
-        sum_value = 0
-        used_coins: Set = set()
-
-        # Use older coins first
-        unspent.sort(reverse=True, key=lambda r: r.coin.amount)
 
         # Try to use coins from the store, if there isn't enough of "unused"
         # coins use change coins that are not confirmed yet
         unconfirmed_removals: Dict[bytes32, Coin] = await self.wallet_state_manager.unconfirmed_removals_for_wallet(
             self.id()
         )
-        for coinrecord in unspent:
-            if sum_value >= amount and len(used_coins) > 0:
-                break
-            if coinrecord.coin.name() in unconfirmed_removals:
-                continue
-            if coinrecord.coin in exclude:
-                continue
-            sum_value += coinrecord.coin.amount
-            used_coins.add(coinrecord.coin)
-            self.log.debug(f"Selected coin: {coinrecord.coin.name()} at height {coinrecord.confirmed_block_height}!")
-
-        # This happens when we couldn't use one of the coins because it's already used
-        # but unconfirmed, and we are waiting for the change. (unconfirmed_additions)
-        if sum_value < amount:
-            raise ValueError(
-                "Can't make this transaction at the moment. Waiting for the change from the previous transaction."
-            )
 
-        self.log.debug(f"Successfully selected coins: {used_coins}")
-        return used_coins
+        coins = await select_coins(
+            spendable_amount,
+            self.wallet_state_manager.constants.MAX_COIN_AMOUNT,
+            spendable_coins,
+            unconfirmed_removals,
+            self.log,
+            uint128(target),
+            exclude,
+        )
+        assert coins is not None and len(coins) > 0
+        assert sum(c.amount for c in coins) >= target
```

Comment:
why are these two asserts here. If these are real errors please Raise something, otherwise just remove them, or if debugging, add debug log entries.
The way I read it is never of these can happen, and if they do it's a serious bug probably worth of a raise of some kind

Reply:
yep just needs to be removed.

Reply:
Re added